src/main/java/org/openstreetmap/atlas/checks/validation/tag/RoadNameGapCheck.java

```diff
@@ -0,0 +1,210 @@
+        {
+            return Optional.empty();
+        }
+
+        final Set<String> matchingInAndOutEdgeNames = getMatchingInAndOutEdgeNames(inEdges,
+                outEdges);
+
+        if (matchingInAndOutEdgeNames.isEmpty())
+        {
+            // There is no pair of inedge and out edge with same name.
+            return Optional.empty();
+        }
+
+        // Create flag when we have in edge and out edge with same name but intermediate edge
+        // doesn't have a name.
+        if (!edge.getName().isPresent())
+        {
+            return Optional.of(
+                    createFlag(object, this.getLocalizedInstruction(0, edge.getOsmIdentifier())));
+        }
+        final Optional<String> edgeName = edge.getName();
+
+        if (edgeName.isPresent() && !matchingInAndOutEdgeNames.contains(edgeName.get()))
+        {
+            final Set<Edge> connectedEdges = edge.connectedEdges().stream()
+                    .filter(this::validCheckForObject).collect(Collectors.toSet());
+            if (findMatchingEdgeNameWithConnectedEdges(connectedEdges, edgeName.get()))
+            {
+                return Optional.empty();
+            }
+            return Optional.of(createFlag(object,
+                    this.getLocalizedInstruction(1, edge.getOsmIdentifier(), edgeName.get())));
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    protected List<String> getFallbackInstructions()
+    {
+        return FALLBACK_INSTRUCTIONS;
+    }
+
+    /**
+     * Compare edge name with individual in and out edges names to see if there is a match. If there
+     * is a match no flag or else flag the edge.
+     *
+     * @param connectedEdges
+     *            connected edges for a given edge
+     * @param edgeName
+     *            Edge name
+     * @return True if there is matching edge name with connected edge or else false.
+     */
+    private boolean findMatchingEdgeNameWithConnectedEdges(final Set<Edge> connectedEdges,
+            final String edgeName)
+    {
+        for (final Edge connectedEdge : connectedEdges)
+        {
+            final Optional<String> connectedEdgeName = connectedEdge.getName();
+            if (connectedEdgeName.isPresent() && connectedEdgeName.get().equals(edgeName))
+            {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Find matching in and out edge names by comparing every name of the edge.
+     *
+     * @param inEdges
+     *            incoming edges
+     * @param outEdges
+     *            outgoing edges
+     * @return Set of matching names for both incoming and outgoing edges.
+     */
+    private Set<String> getMatchingInAndOutEdgeNames(final Set<Edge> inEdges,
+            final Set<Edge> outEdges)
+    {
+        final Set<String> edgeNames = new HashSet<>();
+        for (final Edge inEdge : inEdges)
+        {
+            if (!inEdge.getName().isPresent())
```

Comment:
It would be cleaner if we could add this condition in the line that collects the inEdge and outEdge. 
```edge.inEdges().stream().filter(edge -> edge.getName().isPresent() && this.validCheckForObject(edge) &&....)```